group/assignment5/main.cpp

```diff
@@ -0,0 +1,94 @@
+#include <string>
+#include "test.hpp"
+#include "orgraph.hpp"
+
+std::vector<char> get_alphabet_from_dict (std::vector<std::string> dict)
+{
+    if (dict.empty())
+        return std::vector<char>();
+    std::vector<char> sym_list;
+    std::vector<int> sym_to_id (256, -1);
+    int sym_num = 0;
+    for (std::string word : dict)
```

Comment:
I think that here for each word copy constructor is called?
So this should be enough: for (std::string& word : dict){...}

Reply:
You are absolutely right, I will fix that!